frontend/src/components/organisms/EventForm.tsx

```diff
@@ -14,60 +15,314 @@ import FormControlLabel from "@mui/material/FormControlLabel";
+    ) : null;
+  };
+
+  const CreateSuccessComponent = (): JSX.Element | null => {
+    return successMessage ? (
+      <Alert severity="success">Success: {successMessage}</Alert>
+    ) : null;
+  };
+  /**Helper for handling creating events */
   const handleCreateEvent: SubmitHandler<FormValues> = async (data) => {
-    const {
-      eventName,
-      startDate,
-      endDate,
-      startTime,
-      endTime,
-      location,
-      volunteerSignUpCap,
-      eventDescription,
-      eventImage,
-      rsvpLinkImage,
-    } = data;
-    console.log(data);
+    setIsLoading(true);
+    const mode = status === 0 ? "VIRTUAL" : "IN_PERSON";
+    const startDateTime = convertToISO(getStartTime, getStartDate);
+    const endDateTime = convertToISO(getEndTime, getEndDate);
+    console.log("start time", getStartTime);
+    if (getStartTime == "") {
+      setErrorMessage("Start Time is required.");
+    } else if (getEndTime == "") {
+      setErrorMessage("End Time is required.");
+    } else if (getStartDate == "") {
+      setErrorMessage("Start Date is required.");
+    } else if (getEndDate == "") {
+      setErrorMessage("End Date is required.");
+    } else {
+      var dateValidation =
+        new Date(startDateTime) > new Date(endDateTime) ||
+        new Date(endDateTime) < new Date(startDateTime);
+      if (dateValidation) {
+        setErrorMessage("End Date cannot be earlier than Start Date");
+      } else {
+        setErrorMessage(null);
+      }
+    }
+    const { eventName, location, volunteerSignUpCap, eventDescription } = data;
+
+    const userid = await fetchUserDetails();
+    const fetchCreateEventsUrl = `${url}/events`;
+    try {
+      if (errorMessage) {
+        setIsLoading(false);
+        return;
+      }
+      const createBody = {
+        userID: `${userid}`,
+        event: {
+          name: `${eventName}`,
+          location: `${location}`,
+          description: `${eventDescription}`,
+          startDate: new Date(`${startDateTime}`),
+          endDate: new Date(`${endDateTime}`),
+          capacity: +volunteerSignUpCap,
+          mode: `${mode}`,
+        },
+      };
+      const userToken = await auth.currentUser?.getIdToken();
+      const response = await fetch(fetchCreateEventsUrl, {
+        method: "POST",
+        headers: {
+          Authorization: `Bearer ${userToken}`,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(createBody),
+      });
+      const r = await response.json();
+
+      if (response.ok) {
+        console.log("Successfully Created Event.");
+        setSuccessMessage("Successfully Created Event!");
+        setTimeout(back, 7000);
+      } else {
+        console.error("Unable to Create Event with Status", response.status);
+        console.log("Unable to create event with status", response.status);
+        setErrorMessage(
+          `Unable to create event with status: ${response.status}`
```

Comment:
Please don't show the user a status code. You can use -> We were unable to create this event. Please try again